src/detect_fig.py

```diff
@@ -0,0 +1,203 @@
+"""物体検出を行うモジュール.
+
+ベストショット画像を選択するための物体検出を行う。
+
+NOTE:
+    'learned_weight.pt'は以下にあります。
+    https://drive.google.com/drive/folders/1aUrB3Pj4hzt624SlJv8h1vVOBopAr48l
+    etrobocon2023-camera-system/yolo/の中にダウンロードしてください
+
+参考コード:
+    https://github.com/ultralytics/yolov5
+@author: kawanoichi
+"""
+
+import torch
+from pathlib import Path
+import os
+import numpy as np
+import sys
+from ultralytics.utils.plotting import Annotator, colors
+YOLO_PATH = os.path.join("..", "etrobocon2023-camera-system", "yolo")  # noqa
+sys.path.append(YOLO_PATH)  # noqa
+YOLO_PATH = Path(YOLO_PATH)  # noqa
+from models.common import DetectMultiBackend
+from utils.general import (
+    check_img_size, cv2, non_max_suppression, scale_boxes)
+from utils.torch_utils import select_device
+from utils.augmentations import letterbox
+
+IMAGE_PATH = os.path.join("..", "etrobocon2023-camera-system", "fig_image")  # noqa
+IMAGE_PATH = Path(IMAGE_PATH)  # noqa
+
+
+def check_exists(path) -> None:
+    """ファイル, ディレクトリが存在するかの確認.
+
+    Args:
+        path (str): ファイルまたはディレクトリのパス
+
+    Raises:
+        FileNotFoundError: ファイルが見つからない場合に発生
+    """
+    try:
+        if not os.path.exists(path):
+            raise FileNotFoundError(f"'{path}' does not found\n")
+
+    except FileNotFoundError as e:
+        print("\nError:", e)
+
+
+class Detect():
+    """yolov5(物体検出)をロボコン用に編集したクラス."""
+
+    DEVICE = 'cpu'
+    IMG_SIZE = (640, 480)
+
+    def __init__(self,
+                 img_path=IMAGE_PATH / 'test_image.png',
+                 weights=YOLO_PATH / 'learned_weight.pt',
+                 label_data=YOLO_PATH / 'label_data.yaml',
```

Comment:
もし、フィグ以外の物体検出のことも考えてるなら、ファイル名はフィグ用だとわかる名前がいいと思う。

```suggestion
                 weights=YOLO_PATH / 'fig_learned_weight.pt',
                 label_data=YOLO_PATH / 'fig_label_data.yaml',
```

実際は、figというより、fig_faceなのかな？